scripts/self_update.ts

```diff
@@ -1,15 +1,44 @@
 import Ask from 'ask';
 import iro, { bold, yellow } from 'iro';
+import { difference, format } from 'std/datetime';
 import { dirname, fromFileUrl } from 'std/path';
 
 import { textDecoder } from '/dev_deps.ts';
 import { printLogo } from '/logo/print_logo.ts';
 
 const cwd = dirname(fromFileUrl(import.meta.url));
+const today = format(new Date(), 'yyyy-MM-dd');
+const lastCheckUpdateFileName = 'last_check_update.txt';
 
-checkNewVersion();
+shouldCheckForUpdates();
+
+async function shouldCheckForUpdates() {
+  const checkUpdateIntervalEnv = Deno.env.get('CHECK_UPDATE_INTERVAL');
+
+  if (!checkUpdateIntervalEnv) {
+    return checkNewVersion();
+  }
+
+  try {
+    const lastCheckUpdateDate = new Date(
+      await Deno.readTextFile(
+        lastCheckUpdateFileName,
+      ),
+    );
+
+    const daysSinceLastCheckUpdate = difference(lastCheckUpdateDate, new Date(today), {
+      units: ['days'],
+    }).days;
+
+    daysSinceLastCheckUpdate &&
+      daysSinceLastCheckUpdate >= +checkUpdateIntervalEnv && checkNewVersion();
```

Comment:
where are we using this? It seems to be only conditional without having a variable to hold its value. Shouldn't be better to have a variable and split this into multiple variables to be more semantic?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.svg)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

Reply:
I think you can put an if there

Reply:
ok, I will work on the changes